halo2/src/prover.rs

```diff
@@ -39,7 +40,10 @@ pub fn prove_ast<T: FieldElement>(
 
     let inputs = vec![];
 
-    let params = ParamsKZG::<Bn256>::new(expanded_row_count_log);
+    let params = match params {
+        Some(data) => ParamsKZG::<Bn256>::read(&mut data.as_slice()).unwrap(),
```

Comment:
This actually takes `impl io::Read`, so potentially could pass the file all the way from the CLI to here, instead of collecting into Vec<u8> and reading here.

Reply:
Good idea, these parameters are quite large